deepchem/models/torch_models/layers.py

```diff
@@ -6150,4 +6150,186 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
+    in the rows of `x` and `y`. The two tensors need to have the same number of
+    columns, because one cannot take the dot product between vectors of different
+    lengths. For example, in sentence similarity and sentence classification tasks,
+    the number of columns is the embedding size. In these tasks, the rows of the
+    input tensors would be different test vectors or sentences. The input tensors
+    themselves could be different batches. Using vectors or tensors of all 0s
+    should be avoided.
+
+    Methods
+    -------
+    The vectors in the input tensors are first l2-normalized such that each vector
+    has length or magnitude of 1. The inner product (dot product) is then taken
+    between corresponding pairs of row vectors in the input tensors and returned.
+
+    Examples
+    --------
+    The cosine similarity between two equivalent vectors will be 1. The cosine
+    similarity between two equivalent tensors (tensors where all the elements are
+    the same) will be a tensor of 1s. In this scenario, if the input tensors `x` and
+    `y` are each of shape `(n,p)`, where each element in `x` and `y` is the same, then
+    the output tensor would be a tensor of shape `(n,n)` with 1 in every entry.
+
+    >>> import numpy as np
+    >>> import tensorflow as tf
+    >>> import deepchem.models.layers as layers
+    >>> x = tf.ones((6, 4), dtype=tf.dtypes.float32, name=None)
+    >>> y_same = tf.ones((6, 4), dtype=tf.dtypes.float32, name=None)
+    >>> cos_sim_same = layers.cosine_dist(x,y_same)
+
+    `x` and `y_same` are the same tensor (equivalent at every element, in this
+    case 1). As such, the pairwise inner product of the rows in `x` and `y` will
+    always be 1. The output tensor will be of shape (6,6).
+
+    >>> diff = cos_sim_same - tf.ones((6, 6), dtype=tf.dtypes.float32, name=None)
+    >>> np.allclose(0.0, tf.reduce_sum(diff).numpy(), atol=1e-05)
+    True
+    >>> cos_sim_same.shape
+    TensorShape([6, 6])
+
+    The cosine similarity between two orthogonal vectors will be 0 (by definition).
+    If every row in `x` is orthogonal to every row in `y`, then the output will be a
+    tensor of 0s. In the following example, each row in the tensor `x1` is orthogonal
+    to each row in `x2` because they are halves of an identity matrix.
+
+    >>> identity_tensor = tf.eye(512, dtype=tf.dtypes.float32)
+    >>> x1 = identity_tensor[0:256,:]
+    >>> x2 = identity_tensor[256:512,:]
+    >>> cos_sim_orth = layers.cosine_dist(x1,x2)
+
+    Each row in `x1` is orthogonal to each row in `x2`. As such, the pairwise inner
+    product of the rows in `x1`and `x2` will always be 0. Furthermore, because the
+    shape of the input tensors are both of shape `(256,512)`, the output tensor will
+    be of shape `(256,256)`.
+
+    >>> np.allclose(0.0, tf.reduce_sum(cos_sim_orth).numpy(), atol=1e-05)
+    True
+    >>> cos_sim_orth.shape
+    TensorShape([256, 256])
+
+    Parameters
+    ----------
+    x: tf.Tensor
```

Comment:
please change it to.:
`x: torch.Tensor`

Reply:
changed